css-properties-values-api/Overview.bs

```diff
@@ -601,6 +601,125 @@ referenced, otherwise the declaration is
 
+The <dfn id="at-property-rule">@property</dfn> rule {#at-property-rule}
+===========================================================================
+
+The ''@property'' rule provides an alternative way to register a custom
+property. Well-formed ''@property'' rules result in a
+[=register a custom property|registered custom property=], as if
+{{registerProperty()}} had been called with equivalent parameters.
+
+The semantics of registered properties are the same regardless of
+the mechanism used to perform the registration. This means that, once
+registered, it does not matter whether the registation originated from
+{{registerProperty()}} or ''@property'': the property has the same
+behavior either way.
+
+The syntax of ''@property'' is:
+
+	<pre class="prod def" nohighlight>
+	@property <<custom-property-name>> {
+		<<declaration-list>>
+	}
+	</pre>
+
+The ''@property'' rule accepts the descriptors defined in this chapter.
+
+Issue(845): What happens when we have both an ''@property'' rule, and a
+{{registerProperty()}} call for the same property?
+
+Issue(846): What happens when ''@property'' appears inside a shadow tree?
+
+The 'syntax' descriptor {#syntax-descriptor}
+--------------------------------------------
+
+	<pre class='descdef'>
+	Name: syntax
+	Value: <<string>>
+	For: @property
+	Initial: "*"
+	</pre>
+
+Specifies the syntax of the custom property, on the form defined by
+[[#supported-syntax-strings]]. This descriptor is equivalent to the
+{{PropertyDescriptor/syntax|syntax}} member of {{PropertyDescriptor}}.
+
+The 'inherits' descriptor {#inherits-descriptor}
+------------------------------------------------
+
+	<pre class='descdef'>
+	Name: inherits
+	Value: true | false
+	For: @property
+	Initial: Nothing
+	</pre>
+
+Specifies whether or not the custom property inherits. This is equivalent to the
+{{PropertyDescriptor/inherits}} member of {{PropertyDescriptor}}.
+
+This descriptor is required: if omitted, the ''@property'' rule will not
+register the property.
+
+The 'initial-value' descriptor {#initial-value-descriptor}
+----------------------------------------------------------
+
+	<pre class='descdef'>
+	Name: initial-value
+	Value: <<declaration-value>>
+	For: @property
+	Initial: Nothing
+	</pre>
+
+Specifies the [[css-cascade-4#initial-values|initial value]] of the custom
+property. This is equivalent to the {{PropertyDescriptor/initialValue}} member
+of {{PropertyDescriptor}}.
+
+This descriptor may be omitted when the value of the 'syntax' descriptor is
+<code>"*"</code>, in which case the initial value is
+[[css-variables#invalid-variables|the empty value invalid for substituion]]
+described by [[css-variables]]. Otherwise, the 'initial-value' descriptor is
+required, and the ''@property'' rule will not register the property if it's
+omitted.
+
+The value specified for 'initial-value' must be [=computationally independent=],
+and match the grammar corresponding to the specified 'syntax'.
```

Comment:
What happens if it is not one or the other of these things?

Reply:
Ah, yes good point.

(The whole rule will be invalid).